Add ggp default account retrieval

antonrohr opened the pull request:
This adds orbit_ggp::Account and modifies the creation of orbit_ggp::Client to return the default user account. This will be later used to display a warning when the user connects to an instance of someone else. 

http://b/206103747

beckerhe commented on src/OrbitGgp/ClientTest.cpp:
```diff
@@ -35,6 +35,17 @@ class OrbitGgpClientTest : public testing::Test {
  protected:
+  std::unique_ptr<Client> CreateWorkingClient() {
+    auto create_result = CreateClient(QString::fromStdString(mock_ggp_working_.string()));
+    EXPECT_THAT(create_result, HasValue());
```
ASSERT_THAT?

antonrohr replied:
ASSERT is here not possible, since the macro contains a `return;` and thats incompatible with return type std::unique_ptr<Client>. I am using a CHECK instead. yaqs/q/5581738191355904

beckerhe replied:
Using CHECK is not really a solution because it will kill the process and that means no output XML file will be generated. So this is the same effect as calling `value()` on an ErrorMessageOr instance. (I made the comment mainly to ensure that the test executable won't crash)

beckerhe replied:
Not sure what the best option is in that case.

antonrohr replied:
I removed the whole commit

beckerhe commented on src/OrbitGgp/ClientTest.cpp:
```diff
@@ -35,6 +35,17 @@ class OrbitGgpClientTest : public testing::Test {
+  std::unique_ptr<Client> CreateWorkingClientWith5MsTimeout() {
+    auto create_result = CreateClient(QString::fromStdString(mock_ggp_working_.string()),
+                                      std::chrono::milliseconds{5});
+    EXPECT_THAT(create_result, HasValue());
```
ASSERT_THAT?

antonrohr replied:
see other comment

beckerhe commented on src/OrbitGgp/include/OrbitGgp/Client.h:
```diff
@@ -48,7 +49,7 @@ class Client {
 
 [[nodiscard]] std::chrono::milliseconds GetClientDefaultTimeoutInMs();
-ErrorMessageOr<std::unique_ptr<Client>> CreateClient(
+ErrorMessageOr<std::pair<std::unique_ptr<Client>, Account>> CreateClient(
```
Nit: I think this warrants the creation of a struct containing both the client and the account.

beckerhe replied:
Bot looking at this more closely, I'm not sure anymore if that's the best approach. There are a lot of variables called `client` which should probably be renamed because `client->first` or `client->client` is confusing. Same goes for this function.

Have you considered making the Account retrieval an Async function similar to the `GetWhateverAsync` functions?

beckerhe replied:
Wesome. LGTM!

beckerhe reviewed (approved):
Otherwise LGTM

antonrohr commented:
I rebased and force pushed, since I removed the dedup commit and refactored the Client functionality into `GetDefaultAccountAsync`. It felt like more code was changed then there was in the first round of reviews, so I chose the rebase route
